srtcore/core.cpp

```diff
@@ -1092,11 +1126,58 @@ void CUDT::getOpt(SRT_SOCKOPT optName, void *optval, int &optlen)
         optlen          = sizeof(bool);
         break;
 
-    case SRTO_MAXBW:
+   case SRTO_OUTPACEMODE:
+       *(int *)optval = m_iOutPaceMode;
+       optlen = sizeof(int);
+       if (m_iOutPaceMode == SRT_OPM_UNSET) {
+           //unset return the equivalence based on pre-outpacemode settings
+           if(m_llMaxBW == -1 || m_llMaxBW == BW_INFINITE) {
+               *(int *)optval = SRT_OPM_UNTAMED;
+           }else if (m_llMaxBW > 0) {
+               *(int *)optval = SRT_OPM_CAPPED;
+           }else if (m_llMaxBW == 0) {
+               //Automatic based on set or sampled inputBW
+               if(m_llInputBW > 0)  *(int *)optval = SRT_OPM_INBWSET;
+               else
+               if(m_llInputBW == 0) *(int *)optval = SRT_OPM_SMPINBW;
+           }
+           //SRT_OPM_UNSET returned on invalid/ambiguous setting
+       }
+       break;
+
+   case SRTO_MAXBW:
         *(int64_t *)optval = m_llMaxBW;
         optlen             = sizeof(int64_t);
         break;
 
+    case SRTO_INPUTBW:
+       *(int64_t*)optval = m_llInputBW;
+       optlen             = sizeof(int64_t);
+       break;
+
+   case SRTO_SMPINBW:
+      *(int64_t*)optval = 0LL;
+      if((m_iOutPaceMode == SRT_OPM_SMPINBW)
+      || (m_iOutPaceMode == SRT_OPM_INBWADJ)
+      || ((m_iOutPaceMode == SRT_OPM_UNSET) && (m_llInputBW == 0LL)))
+      {
+	  if(m_pSndBuffer){
+	      //return sampled internally measured input bw
+	      uint64_t llSmpInputBW;
+	      llSmpInputBW = m_pSndBuffer->getInputRate(); //Auto input rate
+	      if( m_pSndBuffer->getInRatePeriod() != 0) { //sampling active
+		  *(int64_t*)optval =llSmpInputBW;
+	      }
+	  }
+      }
+      optlen = sizeof(int64_t);
+      break;
+
+   case SRTO_OHEADBW:
+      *(int *)optval = m_iOverheadBW;
+      optlen = sizeof(int);
+      break;
```

Comment:
Formatting: 4 space alignment.

```suggestion
    case SRTO_OHEADBW:
        *(int *)optval = m_iOverheadBW;
        optlen = sizeof(int);
        break;
```